File: mobile/ios/Quote/Features/Jobs/Models/JobModels.swift
import Foundation

struct Job: Codable, Identifiable {
    let id: String
    let title: String
    let description: String
    let status: String
    let tradeCategory: String
    let budgetMin: Double?
    let budgetMax: Double?
    let preferredStartDate: Date?
    let suburbName: String
    let state: String
    let postcode: String
    let distanceKm: Double
    let quoteCount: Int
    let customerName: String
    let createdAt: Date
    let mediaUrls: [String]

    var budgetDisplay: String {
        if let min = budgetMin, let max = budgetMax {
            return "$\(Int(min)) - $\(Int(max))"
        } else if let min = budgetMin {
            return "From $\(Int(min))"
        } else if let max = budgetMax {
            return "Up to $\(Int(max))"
        }
        return "Negotiable"
    }

    var locationDisplay: String {
        "\(suburbName), \(state) \(postcode)"
    }
}

struct JobDetail: Codable, Identifiable {
    let id: String
    let title: String
    let description: String
    let status: String
    let tradeCategory: TradeCategoryDto
    let budgetMin: Double?
    let budgetMax: Double?
    let preferredStartDate: Date?
    let preferredEndDate: Date?
    let isFlexibleDates: Bool
    let location: LocationDto
    let propertyType: String
    let customer: CustomerDto
    let media: [JobMediaDto]
    let quotes: [QuoteSummaryDto]
    let createdAt: Date
}

struct TradeCategoryDto: Codable {
    let id: String
    let name: String
    let icon: String?
}

struct LocationDto: Codable {
    let latitude: Double
    let longitude: Double
    let suburbName: String
    let state: String
    let postcode: String
}

struct CustomerDto: Codable {
    let id: String
    let firstName: String
    let profilePhotoUrl: String?
}

struct JobMediaDto: Codable, Identifiable {
    let id: String
    let mediaUrl: String
    let mediaType: String
    let caption: String?
    let thumbnailUrl: String?
}

struct QuoteSummaryDto: Codable, Identifiable {
    let id: String
    let tradieId: String
    let tradieName: String
    let totalCost: Double
    let status: String
    let createdAt: Date
}

struct TradeCategory: Codable, Identifiable {
    let id: String
    let name: String
    let description: String?
    let icon: String?
}

struct PaginatedResponse<T: Codable>: Codable {
    let items: [T]
    let pageNumber: Int
    let totalPages: Int
    let totalCount: Int
    let hasPreviousPage: Bool
    let hasNextPage: Bool
}

File: mobile/android/app/src/main/java/com/quote/app/features/jobs/repository/JobModels.kt
package com.quote.app.features.jobs.repository

import java.util.Date

data class Job(
    val id: String,
    val title: String,
    val description: String,
    val status: String,
    val tradeCategory: String,
    val budgetMin: Double?,
    val budgetMax: Double?,
    val preferredStartDate: Date?,
    val suburbName: String,
    val state: String,
    val postcode: String,
    val distanceKm: Double,
    val quoteCount: Int,
    val customerName: String,
    val createdAt: Date,
    val mediaUrls: List<String>
) {
    val budgetDisplay: String
        get() = when {
            budgetMin != null && budgetMax != null -> "$${budgetMin.toInt()} - $${budgetMax.toInt()}"
            budgetMin != null -> "From $${budgetMin.toInt()}"
            budgetMax != null -> "Up to $${budgetMax.toInt()}"
            else -> "Negotiable"
        }

    val locationDisplay: String
        get() = "$suburbName, $state $postcode"
}

data class JobDetail(
    val id: String,
    val title: String,
    val description: String,
    val status: String,
    val tradeCategory: TradeCategoryDto,
    val budgetMin: Double?,
    val budgetMax: Double?,
    val preferredStartDate: Date?,
    val preferredEndDate: Date?,
    val isFlexibleDates: Boolean,
    val location: LocationDto,
    val propertyType: String,
    val customer: CustomerDto,
    val media: List<JobMediaDto>,
    val quotes: List<QuoteSummaryDto>,
    val createdAt: Date
)

data class TradeCategoryDto(
    val id: String,
    val name: String,
    val icon: String?
)

data class LocationDto(
    val latitude: Double,
    val longitude: Double,
    val suburbName: String,
    val state: String,
    val postcode: String
)

data class CustomerDto(
    val id: String,
    val firstName: String,
    val profilePhotoUrl: String?
)

data class JobMediaDto(
    val id: String,
    val mediaUrl: String,
    val mediaType: String,
    val caption: String?,
    val thumbnailUrl: String?
)

data class QuoteSummaryDto(
    val id: String,
    val tradieId: String,
    val tradieName: String,
    val totalCost: Double,
    val status: String,
    val createdAt: Date
)

data class TradeCategory(
    val id: String,
    val name: String,
    val description: String?,
    val icon: String?
)

data class PaginatedResponse<T>(
    val items: List<T>,
    val pageNumber: Int,
    val totalPages: Int,
    val totalCount: Int,
    val hasPreviousPage: Boolean,
    val hasNextPage: Boolean
)
